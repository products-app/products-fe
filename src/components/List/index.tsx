type ListProps = {
  children: React.ReactNode
}

const ListItem = ({ children }: ListProps) => (
  <li className="flex gap-4 py-3 justify-between border-t-2 border-gray-700 border-solid">
    {children}
  </li>
)

const List = ({ children }: ListProps) => {
  return <ul>{children}</ul>
}

export default List
export { ListItem }
