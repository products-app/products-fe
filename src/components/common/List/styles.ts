import { styled } from '@lebernardo/react'

export const LiContainer = styled('li', {
  display: 'flex',
  gap: '1rem',
  padding: '0.75rem 0',
  justifyContent: 'space-between',
  borderTop: '2px solid $gray700',
})
