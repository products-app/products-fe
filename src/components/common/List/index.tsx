// TO DO: move to the design system
import { LiContainer } from './styles'

type ListProps = {
  children: React.ReactNode
}

const ListItem = ({ children }: ListProps) => (
  <LiContainer>{children}</LiContainer>
)

const List = ({ children }: ListProps) => {
  return <ul>{children}</ul>
}

export default List
export { ListItem }
