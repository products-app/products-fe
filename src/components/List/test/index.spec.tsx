import { describe, it, expect } from 'vitest'
import { render } from '@testing-library/react'
import List, { ListItem } from '../index'

const content = <div>Test</div>

describe('List component', () => {
  it('should render correctly', () => {
    const { container, getByText } = render(<List>{content}</List>)

    expect(getByText('Test')).toBeDefined()
    expect(() => getByText('no exist')).toThrow()
    expect(container).toMatchSnapshot()
  })
})

describe('List Item component', () => {
  it('should render correctly', () => {
    const { container, getByText } = render(<List>{content}</List>)

    expect(getByText('Test')).toBeDefined()
    expect(() => getByText('no exist')).toThrow()
    expect(container).toMatchSnapshot()
  })
})

describe('List and List Item components', () => {
  it('should render both components correctly', () => {
    const { container, getByText, getByRole } = render(
      <List>
        <ListItem>{content}</ListItem>
      </List>,
    )

    expect(getByText('Test')).toBeDefined()
    expect(() => getByText('no exist')).toThrow()

    const ulElement = getByRole('list')
    expect(ulElement).toBeInTheDocument()
    const liElement = getByRole('listitem')
    expect(liElement).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })
})
