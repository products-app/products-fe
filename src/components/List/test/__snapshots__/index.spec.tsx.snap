// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Item component should render correctly 1`] = `
<div>
  <ul>
    <div>
      Test
    </div>
  </ul>
</div>
`;

exports[`List and List Item components should render both components correctly 1`] = `
<div>
  <ul>
    <li
      class="flex gap-4 py-3 justify-between border-t-2 border-gray-700 border-solid"
    >
      <div>
        Test
      </div>
    </li>
  </ul>
</div>
`;

exports[`List component should render correctly 1`] = `
<div>
  <ul>
    <div>
      Test
    </div>
  </ul>
</div>
`;
